crate/crash/command.py

```diff
@@ -482,9 +483,26 @@ def _exec_and_print(self, statement: str) -> bool:
         return True
 
 
-def stmt_type(statement):
+def stmt_type(expression: Union[str, sqlparse.sql.Statement]):
     """Extract type of statement, e.g. SELECT, INSERT, UPDATE, DELETE, ..."""
-    return re.findall(r'[\w]+', statement)[0].upper()
+    statement = to_statement(expression)
+    return str(statement.token_first(skip_ws=True, skip_cm=True)).upper()
```

Comment:
That is the spot where the previous naive way of finding the canonical SQL command of an expression was replaced by using a more elaborate method now, completely based on sqlparse's capabilities to skip whitespace and comments.